inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl

```diff
@@ -0,0 +1,429 @@
+                for (uint xi = x_out; xi < x_max_val; ++xi) {
+                    INPUT_VEC input = (INPUT_VEC)(INPUT_INIT_VAL);
+                    #if (REDUCE_MAX_MODE || REDUCE_MIN_MODE || REDUCE_PROD_MODE || REDUCE_AND_MODE || REDUCE_LOG_SUM_EXP_MODE) && REDUCE_FEATURE && (INPUT0_FEATURE_NUM % FSV != 0)
+                        if (fi + get_sub_group_local_id() < INPUT0_FEATURE_NUM)
+                            for (int i = 0; i < READ_OFFSET; ++i)
+                                input[i] = data[offset + get_sub_group_local_id() + i * get_max_sub_group_size()];
+                    #else
+                        input = BLOCK_READ(data, offset);
+                    #endif
+                    unroll_for (int i = 0; i < READ_OFFSET; ++i)
+                        acc[i] = FUNC_CALL(apply_reduce)(acc[i], input[i]);
+                    offset += input_x_pitch;
+                }
+                offset += input_y_pitch - (x_max_val - x_out) * input_x_pitch;
+            }
+            offset += input_fs_pitch - ((y_max_val - y_out) * input_y_pitch);
+        }
+        offset += input_batch_pitch - ((((feature_max_val - feature_out) + FSV - 1) / FSV) * input_fs_pitch);
+    }
+
+    FINAL_ACCUMULATOR_VEC final_acc;
+    unroll_for (uint i = 0; i < READ_OFFSET; ++i) {
+        acc[i] = FUNC_CALL(sub_group_reduce)(acc[i]);
+        final_acc[i] = FUNC_CALL(final_reduce)(TO_FINAL_ACCUMULATOR_TYPE(acc[i]));
+    }
+    
+    OUTPUT_VEC final_result;
+    ACTIVATION_VEC reduce_result = TO_ACTIVATION_VEC(final_acc);
+    #if HAS_FUSED_OPS
+        FUSED_OPS_VECTOR;
+        final_result = FUSED_OPS_RESULT_VECTOR;
+    #else
+        final_result = TO_OUTPUT_VEC(ACTIVATION(reduce_result, ACTIVATION_PARAMS));
+    #endif
+
+    unroll_for (uint i = 0; i < READ_OFFSET; ++i) {
+        if(x + i < COMMON_OUTPUT_SIZE_X){
+            #if REDUCE_BATCH && REDUCE_FEATURE && REDUCE_Y && !KEEP_DIMS
+                output[out_idx + output_y_pitch * i] = final_result[i];
+            #elif REDUCE_FEATURE && REDUCE_Y && !KEEP_DIMS
+                if (get_sub_group_local_id() == 0)
+                    output[out_idx + i] = final_result[i];
+            #elif REDUCE_BATCH && REDUCE_Y && !KEEP_DIMS
+                    output[out_idx + i] = final_result[i];
+            #elif  REDUCE_BATCH && REDUCE_Y && REDUCE_X && !KEEP_DIMS
+                    output[out_idx + get_sub_group_local_id() + output_y_pitch * i] = final_result[i];
+            #elif REDUCE_BFY_BY_FY_Y
+                    output[out_idx + get_sub_group_local_id() + FSV * i] = final_result[i];
+            #elif REDUCE_BATCH && REDUCE_FEATURE && !KEEP_DIMS
+                if (get_sub_group_local_id() == 0)
+                    output[out_idx + i] = final_result[i];
+            #elif REDUCE_BATCH && !KEEP_DIMS
+                    output[out_idx + output_y_pitch * i] = final_result[i];
+            #elif REDUCE_BATCH && REDUCE_FEATURE || REDUCE_BATCH
```

Comment:
Again when there is reduction along features block write shouldn't be used to not corrupt the padding.